pkg/agent/metricsstore/metric_cache.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2020 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricsstore
+
+import (
+	"k8s.io/component-base/metrics"
+)
+
+var (
+	IngressNetworkPolicyCount = metrics.NewGauge(&metrics.GaugeOpts{
```

Comment:
Any reason to exclude these metrics from pkg/agent/metrics/prometheus.go? Or at least from pkg/agent/metrics?

Reply:
Agree with Kobi. These should be part of pkg/agent/metrics IMO.
May be a separate directory like pkg/agent/metrics/networkpolicy will be good. 
In future, we can possibly add a way to enable/disable specific metrics in antrea agent. What do you think @ksamoray ?

Reply:
The reason is to avoid import cycle.
pkg/agent/metrics/prometheus.go use pkg/agent/openflow package
I tried that pkg/agent//openflow/network_policy.go use metrics/prometheus.go, which caused the loop.
What was on my mind was
1. To implement  antrea_agent_ovs_flow_table wihtout using openflow package in prometheus.go
2. Use another package not to have import cycle.
I think 1st one is better considering the future use?

Reply:
Can we do the following to resolve the dependency? Have separate directories for definition and callback (if needed)  for metrics, and just do initialization in prometheus.go
pkg/agent/metrics/ovs --for pod count (ovs container interface count) and ovs flow table count
pkg/agent/metrics/networkpolicy---for network policy related metrics
pkg/agent/metrics/prometheus.go---for initializing/registration of the metrics. In future, we can selectively register the metrics.

Better to get opinion of @ksamoray too.

Reply:
Thank you for your comment. Let me check.

Reply:
I'm guessing that importing pkg/agent/openflow wasn't a good practice on my side.
If possible, I think that it's better to avoid a case where some of the metric implementations import pkg/agent/metrics while others are imported by.
Having said that, it could be painful to reimplement antrea_agent_ovs_flow_table while walking around openflow package. Therefore I think that for now you're solution is valid and so is the proposal by @srikartati 
Unless someone has a different proposal - e.g maintaining flow counts as counter metrics instead of using a callback. My openflow package it too limited to conclude this.

Reply:
As for registering metrics selectively, that would require a whole lot of configuration flags. Maybe a few individual metrics which their collection is costly?

Reply:
@ksamoray I guess it depends on how we group/classify various metrics and interface we provide to enable those groups. Moreover, muxing of metrics is to provide flexibility to the user who is interested in subset of metrics rather than complexity in tracking metrics. 

@yktsubo You probably already saw PR#781. I tried to remove the dependency with other packages. You can now define the new metrics in metrics package. Also, ok with not having separate package for network policy metrics in the current patch. Grouping/classification may probably need more discussion and work.